### PR TITLE
 #  Examples ported to ARM v6 Compiler

### PR DESCRIPTION
**Pull request by massimilian-89**
All the examples now are building on ARM COmpiler V6 , Please check

### CODE REVIEW

**Review by ibenamor — CHANGES_REQUESTED** [review]
The  unused_5xx files should be removed from the project_environment folder. These are no longer useful.

**Review by ibenamor — CHANGES_REQUESTED** [review]
Regression test done. Following examples gives some warnings:
+ ble-Midi
+ ble_burst_adv
+ accel-Sensor
+ ble_Notify_button_Wakeup
+ central_Security_Demo
+ BLE2IR
+ scan_request_track
+simple_beacon
+dynamic_L2CAP_Packet_size_Optimization

Following examples gives some errors:

+MCube-Accel-MC36xx
+segger_rtt

**Comment by koen-vdm** [comment]
I tested all builds for all targets using Keil uVision 5 and I found a few examples not building (see logs attached). 
The following build fails seem to be due to required SDK modifications so these can be ignored:
- connectivity\ibeacon_Optim\project_environment\Power_optimized_dynamic_advertising.uvprojx
- connectivity\scan_request_track\project_enviroment\Scan_request_track.uvprojx
- connectivity\multirole\project_environment\multirole.uvprojx
- connectivity\multi_con_periph\project_envirom
this leaves the following build failure unexplained:
- interfaces\segger_rtt\project_environment\user_segger_rtt_periph.uvprojx

this should be fixed before merging

logs:
[Scan_request_track_log.txt](https://github.com/dialog-semiconductor/BLE_SDK6_examples/files/10921069/Scan_request_track_log.txt)
[user_segger_rtt_periph_log.txt](https://github.com/dialog-semiconductor/BLE_SDK6_examples/files/10921070/user_segger_rtt_periph_log.txt)
[multi_con_periph_log.txt](https://github.com/dialog-semiconductor/BLE_SDK6_examples/files/10921072/multi_con_periph_log.txt)
[multirole_log.txt](https://github.com/dialog-semiconductor/BLE_SDK6_examples/files/10921073/multirole_log.txt)
[Power_optimized_dynamic_advertising_log.txt](https://github.com/dialog-semiconductor/BLE_SDK6_examples/files/10921075/Power_optimized_dynamic_advertising_log.txt)

**Comment by svdpuranik** [comment]
Hi Koen,
multirole, multi_con_periph will not build out of the box so we can igonre them.
user_segger_RTT has errors so change is initiated
Power_optimized has a wrong compiler selected so cahnge is initiazted.
san_request_track is building and has warnings. So should we initiate a change there?

File: interfaces/MCube-Accel-MC36xx/src/platform/user_periph_setup.c [hunk 1]
```diff
@@ -82,7 +82,7 @@ void periph_init(void)
     SetBits16(CLK_16M_REG, XTAL16_BIAS_SH_ENABLE, 1);
 
     //rom patch
-    patch_func();
+//    patch_func();
```
Comment (blauret) [inline interfaces/MCube-Accel-MC36xx/src/platform/user_periph_setup.c:85]: Why? If this is not necessary, please remove.

File: connectivity/BLE2IR/src/keyboard_module/port/port_platform.c [hunk 2]
```diff
@@ -44,7 +44,7 @@ uint32_t port_get_time(void)
 #endif    
 }
 
-__asm void port_delay_usec(uint32_t delay)
+/*__asm void port_delay_usec(uint32_t delay)
```
Comment (blauret) [inline connectivity/BLE2IR/src/keyboard_module/port/port_platform.c:47]: If not necessary, please remove the code.

File: connectivity/hibernation_and_stateaware_hibernation/src/user_hibernation.c [hunk 3]
```diff
@@ -156,7 +156,7 @@ void app_advertise_complete(const uint8_t status)
 		// Close PD_TIM
 		SetBits16(PMU_CTRL_REG, TIM_SLEEP, 1);
 		// Wait until PD_TIM is closed
-		while ((GetWord16(SYS_STAT_REG) & TIM_IS_DOWN) != TIM_IS_DOWN);
+		//while ((GetWord16(SYS_STAT_REG) & TIM_IS_DOWN) != TIM_IS_DOWN);
```
Comment (blauret) [inline connectivity/hibernation_and_stateaware_hibernation/src/user_hibernation.c:159]: Why do we need to change that?

File: interfaces/MCube-Accel-MC36xx/src/user_drivers/platform/console/m_drv_console.c [hunk 4]
```diff
@@ -39,8 +39,8 @@
 void mcube_printf(const char *str, ...)
 {
     /** Function hook by customer. */
-    uart2_write((uint8_t *)str, strlen(str), NULL); // send next string character
-    uart2_finish_transfers();
+    uart_write((uint8_t *)str, strlen(str), NULL); // send next string character
```
Comment (blauret) [inline interfaces/MCube-Accel-MC36xx/src/user_drivers/platform/console/m_drv_console.c:42]: This is a functionnal change. Why is it necesssary

**Review by blauret — CHANGES_REQUESTED** [review]

File: interfaces/MCube-Accel-MC36xx/src/user_drivers/platform/console/m_drv_console.c (hunk 4 above)
Comment (massimilian-89) [reply interfaces/MCube-Accel-MC36xx/src/user_drivers/platform/console/m_drv_console.c:42]: in uart module in SDK the declaration for the uart is :
void uart_write(uint8_t *bufptr, uint32_t size, void (*callback) (uint8_t));
also in config there is this comment :
/****************************************************************************************************************/
/*UART Console Print. Enables serial interface logging mechanism. If CFG_PRINTF is defined CFG_PRINTF_UART2     */
/* controls the uart module used. If it is defined UART2 is used. If not, UART is used. uart or uart2 driver    */
/* must be included in project respectively.                                                                    */
/****************************************************************************************************************/

File: connectivity/hibernation_and_stateaware_hibernation/src/user_hibernation.c (hunk 3 above)
Comment (massimilian-89) [reply connectivity/hibernation_and_stateaware_hibernation/src/user_hibernation.c:159]: solved

File: interfaces/MCube-Accel-MC36xx/src/platform/user_periph_setup.c (hunk 1 above)
Comment (massimilian-89) [reply interfaces/MCube-Accel-MC36xx/src/platform/user_periph_setup.c:85]: solved

**Review by ibenamor — CHANGES_REQUESTED** [review]
Hello @massimilian-89 Could you please have a look to the this warning ?

[Scan_request_track_log.txt](https://github.com/dialog-semiconductor/BLE_SDK6_examples/files/10949427/Scan_request_track_log.txt)

[log.txt](https://github.com/dialog-semiconductor/BLE_SDK6_examples/files/10949426/log.txt)

**Review by ibenamor — APPROVED** [review]
Approve, All the examples are compiling with ARM V6 
Ready to merge

**Review by svdpuranik — APPROVED** [review]
We can merge this Pull Req.

**Pull request merged**